src/components/Skill/Skill.tsx

```diff
@@ -0,0 +1,623 @@
+          ) {
+            return;
+          }
+          onBlur?.(e);
+          if (
+            hoverable &&
+            variant === SkillVariant.Block &&
+            !e.currentTarget.contains(e.relatedTarget)
+          ) {
+            setSkillExpanded(false);
+          }
+        }}
+        onFocus={(e: React.FocusEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onFocus?.(e);
+          if (hoverable && variant === SkillVariant.Block) {
+            setSkillExpanded(true);
+          }
+        }}
+        onKeyDown={(e: React.KeyboardEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onKeyDown?.(e);
+          if (
+            clickable &&
+            expandable &&
+            (e.key === eventKeys.ENTER || e.key === eventKeys.SPACE)
+          ) {
+            if (e.key === eventKeys.SPACE) {
+              e.preventDefault(); // Prevents page scroll
+            }
+            setSkillExpanded((value) => !value);
+          }
+        }}
+        onMouseEnter={(e: React.MouseEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onMouseEnter?.(e);
+          if (hoverable && variant === SkillVariant.Block) {
+            setSkillExpanded(true);
+          }
+        }}
+        onMouseLeave={(e: React.MouseEvent<HTMLDivElement>) => {
+          if (
+            disabled ||
+            readonly ||
+            (variant !== SkillVariant.Block && size === SkillSize.XSmall)
+          ) {
+            return;
+          }
+          onMouseLeave?.(e);
+          if (hoverable && variant === SkillVariant.Block) {
+            setSkillExpanded(false);
+          }
+        }}
+        style={{ ...style, color, height, width }}
+        tabIndex={tabIndex}
+        title={title}
+        ref={mergedRef}
+      >
+        <div
+          className={styles.background}
+          onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+            if (
+              readonly ||
+              disabled ||
+              !clickable ||
+              (variant !== SkillVariant.Block && size === SkillSize.XSmall) ||
+              e.target !== e.currentTarget
+            ) {
+              return;
+            }
+            onClick?.(e);
+            if (expandable && variant === SkillVariant.Block) {
+              setSkillExpanded((value) => !value);
+            }
+          }}
+          style={{ background }}
+        ></div>
+        <div className={styles.content}>{children}</div>
```

Comment:
trying to understand exactly where children fits here. Does children only show if the skill is in block mode, or will it show if it's tag, how would that look?

In the stories, it looks like you're using children to render a slider, and then there's a menu and some other buttons to the right - were those extra pieces part of this component, or custom to your slider, I was not sure

Reply:
The `content` and `expandedContent` can be whatever, but need to be consistent with the layout.

Reply:
And yes, the `Slider` is custom to Storybook, demonstrating one possible scenario.